libraries/chain/controller.cpp

```diff
@@ -301,6 +301,23 @@ struct block_data_t {
       }, v);
    }
 
+   // called from net thread
+	 bool aggregate_vote(const hs_vote_message& vote) {
+	    return std::visit(
+	       overloaded{[](const block_data_legacy_t&) {
+	                     EOS_ASSERT(false, misc_exception, "attempting to call aggregate_vote in legacy mode");
```

Comment:
I think we should just silently return without an exception.. we could be late at switching to IF and receiving votes from other nodes which already switched.